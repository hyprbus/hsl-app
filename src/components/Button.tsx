import * as React from "react";
import styled from "styled-components";

export interface  InterfaceButton {
  action?: () => void;
  className?: string;
  label: string;
}

const BareButton = (props: InterfaceButton) => (
  <div
    className={props.className}
    onClick={props.action}
  >
    {props.label}
  </div>
);

const Button = styled(BareButton)`
  flex: 1;
  cursor: pointer;
  color: ${(props) => props.theme.textColor};
`;

export default Button;
