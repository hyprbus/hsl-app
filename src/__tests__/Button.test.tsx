import * as React from "react";
// import * as TestUtils from "react-addons-test-utils";
import * as renderer from "react-test-renderer";
// import * as Jest from "ts-jest";
import Button from "../components/Button";

describe("Todo component renders the todo correctly", () => {
  it("renders correctly", () => {
    const label = "Test Label";
    const rendered = renderer.create (
      <Button label={label} />,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
