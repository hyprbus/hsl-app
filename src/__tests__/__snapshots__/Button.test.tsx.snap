// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly renders correctly 1`] = `
<div
  className="sc-bdVaJa bmmiQk"
  onClick={undefined}
>
  Test Label
</div>
`;
